File: app/src/main/java/player/phonograph/preferences/HomeTabConfigPreferenceX.java
package player.phonograph.preferences;

import android.content.Context;
import android.util.AttributeSet;

import androidx.annotation.Keep;
import androidx.preference.DialogPreference;

import player.phonograph.R;

@Keep
public class HomeTabConfigPreferenceX extends DialogPreference {
    @Keep
    public HomeTabConfigPreferenceX(Context context) {
        super(context);
        init();
    }

    @Keep
    public HomeTabConfigPreferenceX(Context context, AttributeSet attrs) {
        super(context, attrs);
        init();
    }

    @Keep
    public HomeTabConfigPreferenceX(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
        init();
    }

    @Keep
    public HomeTabConfigPreferenceX(Context context, AttributeSet attrs, int defStyleAttr, int defStyleRes) {
        super(context, attrs, defStyleAttr, defStyleRes);
        init();
    }
    private void init() {
        setLayoutResource(R.layout.x_preference);
    }
}

File: app/src/main/java/player/phonograph/preferences/HomeTabConfigPreferenceX.kt
package player.phonograph.preferences

import android.content.Context
import android.util.AttributeSet
import androidx.annotation.Keep
import androidx.preference.DialogPreference
import player.phonograph.R

@Keep
class HomeTabConfigPreferenceX : DialogPreference {
    @Keep
    constructor(context: Context?) : super(context!!) {
        init()
    }

    @Keep
    constructor(context: Context?, attrs: AttributeSet?) : super(context!!, attrs) {
        init()
    }

    @Keep
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int) : super(context!!, attrs, defStyleAttr) {
        init()
    }

    @Keep
    constructor(context: Context?, attrs: AttributeSet?, defStyleAttr: Int, defStyleRes: Int) : super(
        context!!, attrs, defStyleAttr, defStyleRes) {
        init()
    }

    private fun init() {
        layoutResource = R.layout.x_preference
    }
}